music21/expressions.py

```diff
@@ -1543,81 +1924,210 @@ def name(self) -> str:
 
-        # order matters, see upperPitch and lowerPitch properties below
-        self._otherPitches = (upperPitch, lowerPitch)
+        # order matters, see upperOrnamentalPitch and lowerOrnamentalPitch properties below
+        self._ornamentalPitches = (upperPitch, lowerPitch)
 
     @property
-    def upperPitch(self) -> pitch.Pitch | None:
-        if len(self._otherPitches) >= 1:
-            return self._otherPitches[0]
+    def upperOrnamentalPitch(self) -> pitch.Pitch | None:
+        '''
+        Returns the turn's upper ornamental pitch.  If resolveOrnamentalPitches
+        has not yet been called, None is returned.
+        '''
+        if len(self._ornamentalPitches) >= 1:
+            return self._ornamentalPitches[0]
         return None
 
     @property
-    def lowerPitch(self) -> pitch.Pitch | None:
-        if len(self._otherPitches) >= 2:
-            return self._otherPitches[1]
+    def lowerOrnamentalPitch(self) -> pitch.Pitch | None:
+        '''
+        Returns the turn's lower ornamental pitch.  If resolveOrnamentalPitches
+        has not yet been called, None is returned.
+        '''
+        if len(self._ornamentalPitches) >= 2:
+            return self._ornamentalPitches[1]
         return None
 
-    def realize(self, srcObj: note.Note, *, inPlace=False):
+    def updateAccidentalDisplay(
+        self,
+        *,
+        pitchPast: list[pitch.Pitch] | None = None,
+        pitchPastMeasure: list[pitch.Pitch] | None = None,
+        otherSimultaneousPitches: list[pitch.Pitch] | None = None,
+        alteredPitches: list[pitch.Pitch] | None = None,
+        cautionaryPitchClass: bool = True,
+        cautionaryAll: bool = False,
+        overrideStatus: bool = False,
+        cautionaryNotImmediateRepeat: bool = True,
+    ):
+        '''
+        Updates accidental display for a Turn's ornamental pitches (upper and lower).
+        '''
+        if not self.ornamentalPitches:
+            return
+
+        # upper ornamental pitch
+        p = self.upperOrnamentalPitch
+        if p is not None:
+            if self.upperAccidental is not None and self.upperAccidental.displayStatus is not None:
+                # force upper accidental visibility to be whatever self.upperAccidental's is
+                if p.accidental is None:
+                    p.accidental = pitch.Accidental(0)
+                p.accidental.displayStatus = self.upperAccidental.displayStatus
+            else:
+                p.updateAccidentalDisplay(
+                    pitchPast=pitchPast,
+                    pitchPastMeasure=pitchPastMeasure,
+                    alteredPitches=alteredPitches,
+                    cautionaryPitchClass=cautionaryPitchClass,
+                    cautionaryAll=cautionaryAll,
+                    overrideStatus=overrideStatus,
+                    cautionaryNotImmediateRepeat=cautionaryNotImmediateRepeat,
+                    lastNoteWasTied=False)
```

Comment:
better as a loop:

```
for p, acc in ((self.upperOrnamentalPitch, self.upperAccidental), (self.lowerOrnamentalPitch, self.lowerAccidental)):
    ...
```

then we're saving 15 lines of code.

Reply:
Nice!  Done.